From d8a922ad28c2b8dcdb2e129d249082f96638c1e0 Mon Sep 17 00:00:00 2001
From: Richard Willis <willis.rh@gmail.com>
Date: Thu, 5 Nov 2020 14:49:54 +0000
Subject: [PATCH] Add basicAuth auth header to all headers. Fixes #353 (#354)

---
 lib/openapi3.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/openapi3.js b/lib/openapi3.js
index b3543f1..b41ed0a 100644
--- a/lib/openapi3.js
+++ b/lib/openapi3.js
@@ -279,14 +279,15 @@ function getParameters(data) {
             if (data.api.components.securitySchemes[ess]) {
                 let secScheme = data.api.components.securitySchemes[ess];
                 if (!existingAuth && ((secScheme.type === 'oauth2') || (secScheme.type === 'openIdConnect') ||
-                    ((secScheme.type === 'http') && (secScheme.scheme === 'bearer')))) {
+                    ((secScheme.type === 'http') && ((secScheme.scheme === 'bearer') || (secScheme.scheme === 'basic'))))) {
+                    let method = secScheme.scheme === 'basic' ? 'Basic' : 'Bearer';
                     let authHeader = {};
                     authHeader.name = 'Authorization';
                     authHeader.type = 'string';
                     authHeader.in = 'header';
                     authHeader.isAuth = true;
                     authHeader.exampleValues = {};
-                    authHeader.exampleValues.object = 'Bearer {access-token}';
+                    authHeader.exampleValues.object = method + ' {access-token}';
                     authHeader.exampleValues.json = "'" + authHeader.exampleValues.object + "'";
                     data.allHeaders.push(authHeader);
                 }
-- 
2.18.0

